lib/Controller/Api1Controller.php

```diff
@@ -174,6 +181,82 @@ public function updateSharePermissions(int $shareId, string $permissionType, boo
 		});
 	}
 
+	/**
+	 * @NoAdminRequired
+	 * @CORS
+	 * @NoCSRFRequired
+	 */
+	public function indexViews(int $tableId, string $keyword = null, int $limit = 100, int $offset = 0): DataResponse {
+		if ($keyword) {
+			return $this->handleError(function () use ($keyword, $limit, $offset) {
+				return $this->viewService->search($keyword, $limit, $offset);
+			});
+		} else {
+			return $this->handleError(function () use ($tableId) {
+				return $this->viewService->findAll($this->tableService->find($tableId));
+			});
+		}
+
```

Comment:
Might be personal preference but I feel something like this would be easier to read and saves the extra else branch:

```suggestion
		return $this->handleError(function () use ($keyword, $limit, $offset) use (keyword, $limit, $offset) {
			if ($keyword !== null) {
				return $this->viewService->search($keyword, $limit, $offset);
			}
			
			return $this->viewService->findAll($this->tableService->find($tableId));
		});
```

Null check would also be safer to do a strict comparison as it allows to use the limit/offset with an empty search term. In gerneral findAll and search could probably be combined in one method.

Reply:
Actually looking at it again, i find it odd that search ignores the tableId